pkg/virt-launcher/virtwrap/api/converter_test.go

```diff
@@ -1601,6 +1601,66 @@ var _ = Describe("Converter", func() {
 			Expect(domain.Spec.Devices.Interfaces[0].Source.Bridge).To(Equal("k6t-eth0"))
 			Expect(domain.Spec.Devices.Interfaces[1].Source.Bridge).To(Equal("k6t-net1"))
 		})
+		It("Should create network configuration for macvtap interface and a multus network", func() {
+			v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+			multusNetworkName := "multusNet"
+			networkName := "net1"
+
+			iface1 := v1.Interface{Name: networkName, InterfaceBindingMethod: v1.InterfaceBindingMethod{Macvtap: &v1.InterfaceMacvtap{}}}
+
+			multusNetwork := v1.Network{
+				Name: networkName,
+				NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{NetworkName: multusNetworkName},
+				},
+			}
+			vmi.Spec.Networks = []v1.Network{multusNetwork}
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{iface1}
+
+			domain := vmiToDomain(vmi, c)
+			Expect(domain).NotTo(BeNil())
+			Expect(domain.Spec.Devices.Interfaces).To(HaveLen(1))
+			Expect(domain.Spec.Devices.Interfaces[0].Type).To(Equal("ethernet"), "Macvtap interfaces must be of type `ethernet`")
+		})
+		It("Should create network configuration for macvtap interface and the pod network and a secondary network using multus", func() {
+			v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+			name1 := "net1"
+
+			iface1 := v1.Interface{Name: name1, InterfaceBindingMethod: v1.InterfaceBindingMethod{Macvtap: &v1.InterfaceMacvtap{}}}
+
+			net1 := v1.DefaultPodNetwork()
+			multusNetwork := v1.Network{
+				Name: name1,
+				NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{NetworkName: name1},
+				},
+			}
+			vmi.Spec.Networks = []v1.Network{*net1, multusNetwork}
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface(), iface1}
+
+			domain := vmiToDomain(vmi, c)
+			Expect(domain).NotTo(BeNil())
+			Expect(domain.Spec.Devices.Interfaces).To(HaveLen(2))
+			Expect(domain.Spec.Devices.Interfaces[1].Type).To(Equal("ethernet"), "Macvtap interfaces must be of type `ethernet`")
+		})
+		Specify("macvtap interface binding must be used on a multus network", func() {
+			v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+			name1 := "net1"
+
+			iface1 := v1.Interface{Name: name1, InterfaceBindingMethod: v1.InterfaceBindingMethod{Macvtap: &v1.InterfaceMacvtap{}}}
+
+			podNetwork := v1.Network{
+				Name: name1,
+				NetworkSource: v1.NetworkSource{
+					Pod: &v1.PodNetwork{},
+				},
+			}
+			vmi.Spec.Networks = []v1.Network{podNetwork}
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{iface1}
+
+			domain := &Domain{}
+			Expect(Convert_v1_VirtualMachine_To_api_Domain(vmi, domain, c)).To(HaveOccurred(), "conversion should fail because a macvtap interface requires a multus network attachment")
+		})
```

Comment:
Please add a test to verify the boot order is passed correctly to the domain.

Reply:
Done.